ui/app/AppLayouts/Chat/popups/GroupInfoPopup.qml

```diff
@@ -170,40 +171,35 @@ StatusModal {
         spacing: Style.current.padding
 
         StatusSettingsLineButton {
-            // Not Refactored Yet
-            property int pinnedCount: 0 // popup.store.chatsModelInst.messageView.pinnedMessagesList.count
+            property int pinnedCount: popup.chatContentModule.pinnedMessagesModel.count
 
             id: pinnedMessagesBtn
             visible: pinnedCount > 0
             //% "Pinned messages"
             text: qsTrId("pinned-messages")
             currentValue: pinnedCount
-            onClicked: Global.openPopup(pinnedMessagesPopupComponent, {store: popup.store})
+            onClicked: {
+                popup.store.messageStore.messageModule = popup.chatContentModule.messagesModule
```

Comment:
I see this as a requirement of the PinnedMessagePopup?  Is the update of module instance coupled with user actions?